proto/p4/p4runtime.proto

```diff
@@ -344,6 +344,20 @@ message CounterData {
 }
 
 //------------------------------------------------------------------------------
+// Reserved controller-specified port numbers for reference.
+// Wrapping the enum in a message to avoid name collisions in C++, where "enum
+// values are siblings of their type, not children of it"
+message Ports {
+  enum ReservedPorts {
+    // Port number 0 in invalid. Egress port is initialized to this
+    // controller port number at the beginning of ingress and should
+    // be assigned a valid number in the ingress for all unicast packets.
```

Comment:
PSA v1.0 defines that drop=true for all packets when ingress processing begins, and if you want the packet to be sent unicast to a particular egress port, you must change both drop=false and assign a value to egress_port, which in PSA has no initialized value (i.e. a portable PSA program cannot assume it will be initialized to 0, or any other particular value).  See Section 4.2 of the PSA spec and comments on the metadata fields there.  You are welcome to repeat those details here, but simply leaving this comment as '// Port number 0 is invalid.' might be enough here.